File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/product/event/ProductDefined.java
package cloud.ambar.creditcard.product.event;

import cloud.ambar.common.event.Event;
import lombok.Getter;
import lombok.NonNull;
import lombok.experimental.SuperBuilder;

@SuperBuilder
@Getter
public class ProductDefined extends Event {
    @NonNull private String name;
    @NonNull private Integer interestInBasisPoints;
    @NonNull private Integer annualFeeInCents;
    @NonNull private String paymentCycle;
    @NonNull private Integer creditLimitInCents;
    @NonNull private Integer maxBalanceTransferAllowedInCents;
    @NonNull private String reward;
    @NonNull private String cardBackgroundHex;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Product/Event/ProductDefined.cs
namespace CreditCardEnrollment.CreditCard.Product.Event;

public class ProductDefined : Common.Event.Event
{
    public required string Name { get; init; }
    public required int InterestInBasisPoints { get; init; }
    public required int AnnualFeeInCents { get; init; }
    public required string PaymentCycle { get; init; }
    public required int CreditLimitInCents { get; init; }
    public required int MaxBalanceTransferAllowedInCents { get; init; }
    public required string Reward { get; init; }
    public required string CardBackgroundHex { get; init; }
}
